lib/components/user/nav-login-button.js

```diff
@@ -0,0 +1,124 @@
+import PropTypes from 'prop-types'
+import React, { Component } from 'react'
+import {
+  MenuItem,
+  NavDropdown,
+  NavItem
+} from 'react-bootstrap'
+
+/**
+ * This component displays the sign-in status in the nav bar.
+ * - When a user is not logged in: display 'Sign In' as a link or button.
+ * - When a user is logged in, display an 'avatar' (retrieved from the profile prop)
+ *   and a dropdown button so the user can access more options.
+ */
+export default class NavLoginButton extends Component {
+  static propTypes = {
+    id: PropTypes.string.isRequired,
+    links: PropTypes.arrayOf(PropTypes.shape({
+      target: PropTypes.string,
+      text: PropTypes.string.isRequired,
+      url: PropTypes.string.isRequired
+    })),
+    onSignInClick: PropTypes.func.isRequired,
+    onSignOutClick: PropTypes.func.isRequired,
+    profile: PropTypes.shape({
+      email: PropTypes.string.isRequired,
+      name: PropTypes.string.isRequired,
+      nickname: PropTypes.string,
+      picture: PropTypes.string
+    })
+  }
```

Comment:
@evansiroky, what is your view on using proptypes here? Personally, I favor flow.js over proptypes definitions -- it's much more concise and we actually get the advantage of linting the types (maybe that happens with proptypes, too?). I understand that proptypes was useful for the otp-ui project, but I'm concerned about the typing being actually useful during development.

Reply:
I also prefer static-typing, but think that until we make a commitment to doing that everywhere in otp-rr moving forward, allowing propTypes is better than nothing.